test/tagged-logging-test.ts

```diff
@@ -0,0 +1,34 @@
+import Logger from 'bite-log';
+import { logCountAssert, makeTestPrinter } from './test-helpers';
+
+QUnit.module('Tagged logging');
+
+QUnit.test(
+  'Pushing a tag results in the log mesages being prefixed',
+  assert => {
+    const printer = makeTestPrinter();
+    const logger = new Logger(4, printer);
+    logger.log('hello');          // "hello"
+    logger.log('world');          // "world"
+    logger.pushPrefix('foo');
+    logger.log('goodbye');        // "[foo]: goodbye"
+    logger.log('mars');           // "[foo]: mars"
+    logger.pushPrefix('bar');
+    logger.log('(and saturn!)');  // "[foo][bar]: (and saturn)!"
+    logger.popPrefix();
+    logger.popPrefix();
```

Comment:
What happens if the array is empty and there are no more Prefixes?

Reply:
I have confirmed that it will not be harmful, and consistent with what developers expect from `[].pop();`